addons/hr_holidays/models/hr_leave_allocation.py

```diff
@@ -575,7 +575,7 @@ def _get_future_leaves_on(self, accrual_date):
         if self.type_request_unit in ['hour']:
             return float_round(fake_allocation.number_of_hours_display - self.number_of_hours_display, precision_digits=2)
         res = round((fake_allocation.number_of_days - self.number_of_days), 2)
-        self._invalidate_cache()
+        fake_allocation._invalidate_cache(['number_of_days', 'number_of_days_display', 'lastcall', 'nextcall'])
```

Comment:
```suggestion
        fake_allocation._invalidate_cache()
```

Reply:
I have tested all your suggestion, not specifying the fields doesn't work and same when calling onchange_date_from.  This is really the only change I managed to make it work in.

I have added the display_hours field and separated the test for result persistance from the rest.